generatorreceiver/internal/topology/topology.go

```diff
@@ -35,12 +35,12 @@ func (t *Topology) traverseServiceGraph(service string, route string, seenCalls
 	downstreamCalls := t.GetServiceTier(service).GetRoute(route).DownstreamCalls
 	// already validated existence of all services/routes, so ^ is safe
 	seenCalls[service+route] = true
-	for ds, dr := range downstreamCalls {
-		if seenCalls[ds+dr] {
-			return fmt.Errorf(printServiceCycle(orderedCalls, ds+dr))
+	for _, r := range downstreamCalls {
```

Comment:
nit: same as my previous comment, you can change this loop variable `r` to `call` or `c` or `dc`